type-equality-compat/type-equality-compat.cabal

```diff
@@ -0,0 +1,44 @@
+name:          type-equality-compat
+version:       1
+stability:     provisional
+cabal-version: >=1.10
+build-type:    Simple
+author:        Oleg Grenrus <oleg.grenrus@iki.fi>
+maintainer:    Oleg Grenrus <oleg.grenrus@iki.fi>
+license:       BSD3
+license-file:  LICENSE
+homepage:      https://github.com/phadej/some
+category:      Data, Dependent Types
+synopsis:      Data.Type.Equality compat package
+description:
+  This library defines a propositional equality data type,
+  shims @Data.Type.Equality" as well as possible for older GHCs (< 7.8).
+  .
+  @
+  data a :~: b where
+  \    Refl :: a :~: a
+  @
+  .
+  The module @Data.Type.Equality.Hetero@ shims @:~~:@ equality, for
+  compilers with @PolyKinds@
+
+tested-with:
+  GHC ==8.8.1 || ==8.6.5 || ==8.4.4 || ==8.2.2 || ==8.0.2 || ==7.10.3 || ==7.8.4 || ==7.6.3 || ==7.4.2 || ==7.2.2 || ==7.0.4
+
+source-repository head
+  type:     git
+  location: git://github.com/phadej/some.git
+  subdir:   type-equality-compat
+
+library
+  default-language: Haskell2010
+  build-depends:    base >=4.3 && <4.14
+
+  if !impl(ghc >=7.8)
+    hs-source-dirs: src-old
+    exposed-modules:  Data.Type.Equality
+
+  if impl(ghc >=8.0 && <8.2)
```

Comment:
This is what I really meant to comment on. Is there any reason to restrict exposing `Data.Type.Equality.Hetero` to only GHC <8.2? Why not just 8.0 and onwards?

Reply:
Good catch!